### PR TITLE
grpcproxy: add SetHeader support to ServerStream

### LINKED ISSUE #6726: etcd : Go get command giving interface implementation error for wserver
image

### PR DESCRIPTION
**Pull request by heyitsanthony**
Fixes #6726

### CODE REVIEW

File: proxy/grpcproxy/watch_client_adapter.go [hunk 1]
```diff
@@ -73,17 +75,34 @@ type chanServerStream struct {
...
+		ss.headerc = nil
 		return nil
 	case <-ss.Context().Done():
 	}
 	return ss.Context().Err()
 }
 
+func (ss *chanServerStream) SetHeader(md metadata.MD) error {
+	if ss.headerc == nil {
+		return fmt.Errorf("chanServeStream: already sent header")
```
Comment (xiang90) [inline proxy/grpcproxy/watch_client_adapter.go]: errors.New()?

File: proxy/grpcproxy/watch_client_adapter.go [hunk 2]
```diff
@@ -73,17 +75,34 @@ type chanServerStream struct {
...
+	outmd := make(map[string][]string)
+	for _, h := range append(ss.headers, md) {
+		for k, v := range h {
+			outmd[k] = v
+		}
+	}
 	select {
-	case ss.headerc <- md:
+	case ss.headerc <- outmd:
+		ss.headerc = nil
```
Comment (xiang90) [inline proxy/grpcproxy/watch_client_adapter.go]: sending the 2nd header should block? is this doced somewhere?

**Comment by xiang90** [comment]
lgtm

File: proxy/grpcproxy/watch_client_adapter.go (hunk 2 above)
Comment (heyitsanthony) [inline proxy/grpcproxy/watch_client_adapter.go]: oh, yeah it should probably return an error; it should only be called once according to the docs

**Pull request merged**